eq-author/src/components/BaseLayout/PermissionsBanner.js

```diff
@@ -6,18 +6,30 @@ import PropTypes from "prop-types";
 import IconText from "components/IconText";
 import ExpansionTransition from "components/transitions/ExpansionTransition";
 import WarningIcon from "components/OfflineBanner/icon-warning.svg?inline";
+import Button from "components/buttons/Button";
 
 import { colors } from "constants/theme";
 import { READ, WRITE } from "constants/questionnaire-permissions";
 import { AWAITING_APPROVAL } from "constants/publishStatus";
 
+import { useQuestionnaireLockingModal } from "components/modals/QuestionnaireLockingModal";
+
 const Banner = styled.div`
   background-color: ${colors.red};
   height: 2.5em;
   justify-content: center;
   display: flex;
 `;
 
+const UnlockButton = styled(Button).attrs(() => ({
+  small: true,
+  variant: "tertiary-light",
+}))`
+  margin-left: 1em;
+  padding: 0.25em 0.5em;
+  border: 1px solid white;
```

Comment:
```suggestion
  border: 1px solid ${colors.white};
```

Reply:
Cheers Matt - have pushed colour constant change.

And aye, thought it was kind of neat to hide away all the other stuff that otherwise leaks out (e.g. parents have to manually open / close modals and maintain their state even though it's really not part of their jurisdiction). Can discuss at tech session sure.

Would be cool if we could make a generic hook in that style for all similar modals - would just have to standardise the prop names so it's always `onConfirm` and `onCancel` etc.